test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -167,7 +167,7 @@ object Assertion extends AssertionVariants {
    * Makes a new assertion that requires an exit value to die with an instance
    * of given type (or its subtype).
    */
-  def diesWithA[E: ClassTag]: Assertion[Exit[E, Any]] =
+  def diesWithA[E <: Throwable : ClassTag]: Assertion[Exit[Any, Any]] =
```

Comment:
I don't think you need the type constraint here.

Reply:
Thanks for the comment. I have fixed that